### PR TITLE
Let to update all configurable fields of the main part of an AEA configuration.

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

Let to update all configurable fields of the main part of an `AgentConfig` configuration via `AgentConfig.update`.

## Fixes

n/a

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

n/a

### CODE REVIEW

File: aea/configurations/base.py [hunk 1]
```diff
@@ -1703,7 +1705,14 @@ def update(self, data: Dict) -> None:
         self.component_configurations = result
 
         # update other fields
-        # currently not supported.
+        for item_id, value in data.get("private_key_paths", {}).items():
+            self.private_key_paths.update(item_id, value)
```
Comment (marcofavorito) [inline aea/configurations/base.py:1709]: is it correct we only update existing keys?

File: aea/configurations/base.py [hunk 2]
```diff
@@ -1703,7 +1705,14 @@ def update(self, data: Dict) -> None:
...
+
+        for item_id, value in data.get("connection_private_key_paths", {}).items():
+            self.connection_private_key_paths.update(item_id, value)
+
+        self.logging_config = data.get("logging_config", self.logging_config)
```
Comment (marcofavorito) [inline aea/configurations/base.py:1714]: here we don't apply granular set... I think it's OK

File: aea/configurations/base.py (hunk 1 above)
Comment (DavidMinarsch) [reply aea/configurations/base.py:1709]: No, at the moment we need to be able to add and override, otherwise it is not very useful

**Review by DavidMinarsch — CHANGES_REQUESTED** [review]

**Review by DavidMinarsch — APPROVED** [review]
LGTM

File: aea/configurations/base.py (hunk 1 above)
Comment (marcofavorito) [reply aea/configurations/base.py:1707]: OK, the behaviour of `CRUDCollection.update` now creates an item if it the key doesn't exists yet.

**Review by solarw — APPROVED** [review]
LGTM

**Pull request merged**